### PR TITLE
Tutorial path vectors  and minimal cut sets 

### PR DESCRIPTION
**Pull request by syarra**
Thanks Susan for this contribution.

**I hereby confirm that I have:**

- [x] Tested my code on my own machine
- [x] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [x] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

**Comment by rmtfleming** [comment]
@syarra model.rev is depreciated
Please add citation to both new function interfaces
Klamt S, Saez-Rodriguez J and Gilles ED (2007) Structural and functional analysis of cellular networks with CellNetAnalyzer. BMC Systems Biology 1: 2
Also add pointer to download of CNA
http://www2.mpi-magdeburg.mpg.de/projects/cna/download.html

File: src/modelAnalysis/topology/minimalCutSets.m [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+%
+%    output = minimialCutSets(targets, varargin)
+%
+% INPUT:
+%    targets (mandatory): a binary matrix that row-wise contains the
+%                         target paths/cycles/elementary modes; a '1' in
+%                         the i-th row and j-th column in targets indicates
+%                         participation of element (reaction) j in mode i;
+%
+%    mcsmax:              maximal size of cutsets to be calculated; must
```
Comment (tpfau) [inline src/modelAnalysis/topology/minimalCutSets.m]: This should be splitted into optional and required inputs.

File: src/modelAnalysis/topology/minimalCutSets.m [hunk 2]
```diff
@@ -0,0 +1,93 @@
+function [output] = minimalCutSets(targets, varargin)
+% Computes minimal cut sets for paths/cycles/elementary modes with Berge algorithm
```
Comment (tpfau) [inline src/modelAnalysis/topology/minimalCutSets.m]: The signature has to be changed to use optional arguments and not parameter/value pairs as required by @rmtfleming in #497

File: src/modelAnalysis/topology/pathVectors.m [hunk 3]
```diff
@@ -0,0 +1,195 @@
+function [output] = pathVectors(model, directory, varargin)
+% Computes elementary mode and extreme pathway
```
Comment (tpfau) [inline src/modelAnalysis/topology/pathVectors.m]: The signature has to be changed to use optional arguments and not parameter/value pairs as required by @rmtfleming in #497

File: src/modelAnalysis/topology/pathVectors.m [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    error('iso_flag should be either 0 or 1');
+end
+
+if positivity ~= 0 && positivity ~= 1
+    error('positivity should be either 0 or 1');
+end
+
+%% Convertining reversible reactions into irreversible reactions
+if positivity
+    ind = model.rev == 1;
```
Comment (tpfau) [inline src/modelAnalysis/topology/pathVectors.m]: as mentioned b< @rmtfleming : should be model.lb < 0 and .rev should be ignored (its also not checked by CellNetAnalyser)

File: src/modelAnalysis/topology/pathVectors.m [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    rev = [model.rev; model.rev(ind)];
+    rxns = strcat(model.rxnNames(ind), '-rev');
+    c = [model.c; model.c(ind)];
```
Comment (tpfau) [inline src/modelAnalysis/topology/pathVectors.m]: CellNetAnalyser assumes that the model objective in a COBRA model is always maximisation. 
We have to either coordinate with the CNA authors, that they consider the osense field, or we have to modify the csense according to the osense field.

File: src/modelAnalysis/topology/minimalCutSets.m (hunk 1 above)
Comment (syarra) [inline src/modelAnalysis/topology/minimalCutSets.m]: Done in 0f9ec34.

File: src/modelAnalysis/topology/pathVectors.m (hunk 4 above)
Comment (syarra) [inline src/modelAnalysis/topology/pathVectors.m]: Done in 0f9ec34.

**Pull request merged**